rtos/Thread.h

```diff
@@ -347,6 +347,7 @@ class Thread {
     bool _dynamic_stack;
     Semaphore _join_sem;
     Mutex _mutex;
+    bool _finished;
```

Comment:
Random thought: Should we be ifdeffing these variables out or something for non-debug builds? Otherwise they are just wasting space.